File: Exercism Codes/Java/Bob.java
import java.util.regex.Pattern;

public class Bob {

    public String hey(String phrase) {
        phrase = phrase.trim();

        if(phrase.endsWith("?") &&
                phrase.toUpperCase().equals(phrase) &&
                Pattern.compile(".*[a-zA-Z].*").matcher(phrase).matches()
        ) {
            return "Calm down, I know what I'm doing!";
        }
        else if(phrase.endsWith("?")) {
            return "Sure.";
        }
        else if(phrase.equals("")) {
            return "Fine. Be that way!";
        }
        else if(phrase.toUpperCase().equals(phrase) &&
                Pattern.compile(".*[a-zA-Z].*").matcher(phrase).matches()
        ) {
            return "Whoa, chill out!";
        }
        else {
            return "Whatever.";
        }
    }
}

File: Exercism Codes/Kotlin/Bob.kt
object Bob {
    fun hey(input: String) = when {
        questionMark(input) && isUppercase(input) -> "Calm down, I know what I'm doing!"
        questionMark(input) -> "Sure."
        isUppercase(input) -> "Whoa, chill out!"
        input.trim().isEmpty() -> "Fine. Be that way!"
        else -> "Whatever."
    }
    private fun questionMark(input: String) = input.trim().endsWith("?")
    private fun isUppercase(input: String) = input.uppercase() == input && "[A-Za-z]".toRegex().containsMatchIn(input)
}